### PR TITLE
NETOBSERV-759: decorate flows with agent IP

### PR DESCRIPTION
**Pull request by mariomac**
Only for Protocol Buffers encoding. IPFIX is still missing.

This PR creates a new "Decorator" instance and moves there also the interface naming process.

At least for OVN-K, the default `external` IP parsing coincides with the Node IP without any kind of tuning from the NOO:
image

### CODE REVIEW

**Comment by eranra** [comment]
@mariomac can you provide some motivation for this enrichment :-) --- we have the network Interface ID --- can't we get that data from k8s and not send that field over and over for each flow?

**Comment by jotak** [comment]
@eranra do you have strong concerns about this?
It's really solving a lot of issues that we have, regarding the interpretation of `FlowDirection` (what we assume on frontend side != what it is at the low level, and knowing the Agent IP was the missing piece between them)

**Comment by jotak** [comment]
tested and works as expected
@praveingk do you want to review? You know better the code than me :)

File: pkg/exporter/ipfix.go [hunk 1]
```diff
@@ -12,6 +12,9 @@ import (
 
 var ilog = logrus.WithField("component", "exporter/IPFIXProto")
 
+// TODO: encode also the equivalent of the Protobuf's AgentIP field in a format that is binary-
```
Comment (jotak) [inline pkg/exporter/ipfix.go:15]: For information, in goflow's IPFIX this was called "SamplerAddress": https://github.com/cloudflare/goflow#output-format

In IPFIX spec (https://www.iana.org/assignments/ipfix/ipfix.xhtml) this is `exporterIPv4Address` / `exporterIPv6Address`

File: pkg/flow/decorator.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package flow
+
+import (
+	"net"
+)
+
+type InterfaceNamer func(ifIndex int) string
+
+// Decorate adds to the flows extra metadata fields that are not directly fetched by eBPF:
+// - The interface name (corresponding to the interface index in the flow).
```
Comment (jotak) [inline pkg/flow/decorator.go:10]: @mariomac the interface name was already in the flows, is it just that you refactored at the same time the way it is added, right? Or did you change the flow content wrt interface?

**Review by jotak — APPROVED** [review]
thanks!

File: pkg/flow/decorator.go (hunk 2 above)
Comment (mariomac) [reply pkg/flow/decorator.go:10]: Yes, it was added in an earlier stage in the flow creation, but the constructor required to inject an external component (interfaceNamer) and I think that moving the extra decoration to another stage makes the code cleaner.

**Comment by mariomac** [comment]
Merging to unblock other PRs/tests and, if @praveingk has any concern about the current PR, we can address them into another PR.

File: pkg/exporter/ipfix.go (hunk 1 above)
Comment (praveingk) [reply pkg/exporter/ipfix.go:15]: @jotak the ExportingProcess has a net.Conn, which could be used to get the info. Yes, exporterIPv4(/6)Address seems right though

Comment (praveingk) [reply pkg/exporter/ipfix.go:15]: However, the ExportingProcess could be used only if the AgentIPIface is set as "External"

**Pull request merged**